ArduPlane/mode_takeoff.cpp

```diff
@@ -0,0 +1,114 @@
+#include "mode.h"
+#include "Plane.h"
+
+/*
+  mode takeoff parameters
+ */
+const AP_Param::GroupInfo ModeTakeoff::var_info[] = {
+    // @Param: ALT
+    // @DisplayName: Takeoff mode altitude
+    // @Description: This is the target altitude for TAKEOFF mode
+    // @Range: 0 200
+    // @Increment: 1
+    // @Units: m
+    // @User: User
+    AP_GROUPINFO("ALT", 1, ModeTakeoff, target_alt, 50),
+
+    // @Param: LVL_ALT
+    // @DisplayName: Takeoff mode altitude level altitude
+    // @Description: This is the altitude below which wings are held level for TAKEOFF mode
+    // @Range: 0 50
+    // @Increment: 1
+    // @Units: m
+    // @User: User
+    AP_GROUPINFO("LVL_ALT", 2, ModeTakeoff, level_alt, 20),
+
+    // @Param: LVL_PITCH
+    // @DisplayName: Takeoff mode altitude initial pitch
+    // @Description: This is the target pitch for the initial climb to TKOFF_LVL_ALT
+    // @Range: 0 30
+    // @Increment: 1
+    // @Units: deg
+    // @User: User
+    AP_GROUPINFO("LVL_PITCH", 3, ModeTakeoff, level_pitch, 15),
+
+    // @Param: DIST
+    // @DisplayName: Takeoff mode distance
+    // @Description: This is the distance from the takeoff location where the plane will loiter. The loiter point will be in the direction of takeoff (the direction the plane is facing when the motor starts)
+    // @Range: 0 500
+    // @Increment: 1
+    // @Units: m
+    // @User: User
+    AP_GROUPINFO("DIST", 4, ModeTakeoff, target_dist, 200),
```

Comment:
This needs to be lower. I'm thinking 100. Default loiter radius is 60.

Reply:
disagree, many flying fields don't like autonomous planes flying anti-pattern in the pattern.  I think it's best to let this go a little further out by default, and let the user bring it in if their condition allows.  Just my $0.02

Reply:
by default, the plane will be orbiting with a min distance of 200+/-60 plus the 50 altitude. That's almost an effective 300m distance. That's a bit crazy. If I threw my plane up I would think it's flying away!!!!

Reply:
I flew this today.  Having the plane orbit with its furthest point away at almost 300m was not a concern, and also kept it out of the most immediate flying space where some others were toying around.